rust_src/src/data.rs

```diff
@@ -779,4 +779,16 @@ pub fn get_variable_watchers(symbol: LispSymbolRef) -> LispObject {
     }
 }
 
+/// Return population count of VALUE.
+/// This is the number of one bits in the two's complement representation
+/// of VALUE.  If VALUE is negative, return the number of zero bits in the
+/// representation.
+#[lisp_fn]
+pub fn logcount(mut value: EmacsInt) -> i32 {
```

Comment:
Why not mimic the C and shadow `value` instead of using a `mut`?

```
let value = if value < 0 { -1 - value } else { value };
unsafe { rust_count_one_bits(value as usize) }
```